TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,591 @@
+                    telemetry.addData("Goal Degrees", goalDegrees);
+                    telemetry.addData("Current Degrees", getYaw());
+                    return false;
+                } else {
+                    this.stop();
+                    goalDegrees = -1;
+                    return true;
+                }
+            case TURNRIGHT:
+                if (goalDegrees == -1) {
+                    goalDegrees = (this.getYaw() + degrees);
+                    if (goalDegrees > 360) {
+                        goalDegrees = (goalDegrees - 360);
+                    }
+                }
+                if (!(this.getYaw() > (goalDegrees - gyroRange) && this.getYaw() < (goalDegrees + gyroRange))) {
+                    drive(direction, power);
+                    telemetry.addData("Goal Degrees", goalDegrees);
+                    telemetry.addData("Current Degrees", getYaw());
+                    return false;
+                } else {
+                    this.stop();
+                    goalDegrees = -1;
+                    return true;
+                }
+        }
+        return false;
+    }
+
+    public float getYaw(){
+        if(this.gyro.getYaw() < 0) {
+            return (360 + this.gyro.getYaw());
+        } else {
+            return this.gyro.getYaw();
+        }
+    }
+
+    //Experiment with using a field oriented drive instead of a robot oriented drive
+    public void experimentalDrive(double x, double y, double z){
+        /*
+        Guide to motor Powers:
+        Left Front: - (y + x + Z)
+        Right Front: y - x - z
+        Left Rear: y + x - z
+        Right Rear: - (y - x + z)
+         */
+
+        double forward = x; //Invert stick X axis
+        double strafe = y;
+
+        double pi = Math.PI;
+
+        /* Adjust Joystick X/Y inputs by navX MXP yaw angle */
+
+        double gyro_radians = getYaw() * pi/180;
+        double temp = forward * cos(gyro_radians) + strafe * sin(gyro_radians);
+        strafe = -forward * sin(gyro_radians) + strafe * cos(gyro_radians);
+        forward = temp;
+
+        /* At this point, Joystick X/Y (strafe/forward) vectors have been */
+        /* rotated by the gyro angle, and can be sent to drive system */
+
+        this.leftFront.setPower(Range.clip(-(forward+strafe+z),-1,1));
+        this.rightFront.setPower(Range.clip((forward-strafe-z),-1,1));
+        this.leftRear.setPower(Range.clip(-(forward-strafe+z),-1,1));
+        this.rightRear.setPower(Range.clip((forward+strafe-z), -1, 1));
+    }
+
+    public double currentDegrees(double x, double y) {
+        if (((Math.atan2(y, x)) * (180/Math.PI))<0) {
+            return (360 + ((Math.atan2(y, x)) * (180/Math.PI)));
+        } else {
+            return ((Math.atan2(y, x)) * (180/Math.PI));
+        }
+    }
+
+    public double currentZone(double x, double y) {
+        if (((int)(Math.round(currentDegrees(x, y) / 22.5))) <= 15) {
+            return (int)(Math.round(currentDegrees(x, y) / 22.5));
```

Comment:
#fix let's save `(int)(Math.round(currentDegrees(x, y)` into a variable so we don't need to call it twice.